jupyterhub/apihandlers/users.py

```diff
@@ -18,6 +18,41 @@ def get(self):
         users = self.db.query(orm.User)
         data = [ self.user_model(u) for u in users ]
         self.write(json.dumps(data))
+    
+    @admin_only
+    @gen.coroutine
+    def post(self):
+        data = self.get_json_body()
+        if not data or not isinstance(data, dict) or not data.get('usernames'):
+            raise web.HTTPError(400, "Must specify at least one user to create")
+        
+        usernames = data.pop('usernames')
+        admin = data.get('admin', False)
+        self._check_user_model(data)
+        
+        for name in usernames:
+            user = self.find_user(name)
+            if user is not None:
+                raise web.HTTPError(400, "User %s already exists" % name)
+        
+        created = []
+        for name in usernames:
+            user = self.user_from_username(name)
+            if admin:
```

Comment:
Mmh, admin was pulled globally *once* above, but now it will be applied to all users? Either I'm misunderstanding the logic, or it's wrong...